EngineLayer/MetaMorpheusEngine.cs

```diff
@@ -80,14 +80,14 @@ public static List<MatchedFragmentIon> MatchFragmentIons(Ms2ScanWithSpecificMass
         {
             var matchedFragmentIons = new List<MatchedFragmentIon>();
 
-            // if the spectrum has no peaks
-            if (!scan.ExperimentalFragments.Any())
-            {
-                return matchedFragmentIons;
-            }
-
             if (scan.TheScan.MassSpectrum.XcorrProcessed)
             {
+                // if the spectrum has no peaks
```

Comment:
We should keep this outside of the if statement, because it needs to be run regardless.

Reply:
I think we need to keep it as checking scan.ExperimentalFragments, because the matching happens to deconvolutedMasses (private), which is related more with experimentalFragments than with raw xArray peaks.

Reply:
there's a check after the lowCID stuff which checks if there are any deconvoluted masses

Reply:
(line 111)